### PR TITLE
✨ Feature: Dependency-diff API optimize: var re-naming, removing unused JSON tags

### PR DESCRIPTION
**Pull request by aidenwang9867**
#### What kind of change does this PR introduce?

1. `GetDependencydiffResults` API (PR #2046) input param optimization (renaming ambiguous ones).

2. Remove unnecessary struct JSON tags for `DependencyCheckResult`: will define a `JSONDependencyCheckResult` struct with JSON tags in PR #2077.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
Var names such as `baseSHA` and `headSHA` might be confusing and ambiguous since users can use either SHAs or branch names as the inputs; changing them to `base` and `head`. There are some other minor changes on var names.

Struct JSON tags for `DependencyCheckResult` are unnecessary since we will define a `JSONDependencyCheckResult` struct with JSON tags in PR #2077 to output JSON. (removing these tags)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]
LGTM, thanks

File: pkg/dependencydiff_result.go [hunk 1]
```diff
@@ -44,40 +44,41 @@ func (ct *ChangeType) IsValid() bool {
...
 	// Error is an error returned when running the scorecard checks. A nil Error indicates the run succeeded.
-	Error error `json:"scorecardRunTimeError"`
+	Error error
 }
 
 // DependencyCheckResult is the dependency structure used in the returned results.
 type DependencyCheckResult struct {
 	// ChangeType indicates whether the dependency is added, updated, or removed.
-	ChangeType *ChangeType `json:"changeType"`
+	ChangeType *ChangeType
```
Comment (laurentsimon) [inline pkg/dependencydiff_result.go:60]: I'm curious why the JSON fields are no longer need

Comment (aidenwang9867) [reply pkg/dependencydiff_result.go:60]: We'll have a `JSONDependencyCheckReuslt` in PR #2077 to output JSON, like `ScorecardResult` and `JSONScorecardResultV2`, so IMO, we don't need json tags for `DependencyCheckResult`, is that right?

Comment (laurentsimon) [reply pkg/dependencydiff_result.go:60]: ah, you mean this struct is copied into another one. Got it.

**Pull request merged**